Does  your code contain the mask mAP computation?

wuleibupt opened the pull request.

souryuu committed:
added mask visualization
added test.py

souryuu committed:
fixed multiple mask related issue
- bad assignment in mask.py
- smoothen the cropped features from pyramid in crop.py
- other visualization and minor parameter tuning

fixed network setting
- add relu and maxpooling pad to non-batch normalization scope

souryuu committed:
remove some comments

souryuu committed:
Changes in a network
- remove batch normalization in the layers before loss

Fixed some mask issues
- temporary fixed inaccurate gt_mask from cv crop and resize
- change gt_mask from int to float
- smoothen mask by cv INTER_CUBIC

Misc.
- match all rois and their target (manually checked through indexs)

souryuu committed:
Changed training mask sampling method and IoU threshold

souryuu committed:
fixed some indention

souryuu committed:
clean up variable names and comments

souryuu committed:
fixed test.py
corrected mask losses in pyramid_network.py

souryuu committed:
merged inst.py to sample.py
changed some variable names

souryuu committed:
merge to master

souryuu committed:
clean up comments
added color codes

souryuu committed:
comments

souryuu committed:
fixed only_positive in sample_rpn_outputs

souryuu committed:
fixed only_positive in sample_rpn_outputs_wrt_gt

souryuu committed:
changed some hyper params

souryuu committed:
remove gt during testing

souryuu committed:
fixed nms in sampling during test

souryuu committed:
fixed conflict from variable names

souryuu committed:
last check before change anchor

souryuu committed:
changed anchor from 3x3 to 5x3

souryuu committed:
changed anchor to match with MaskRCNN original paper
fixed the issue that is_training=False causes no bounding boxes during test
fixed batch normalization gradient update

souryuu committed:
speed up sorting in sample_rpn_outputs
change anchor back to 3x3 (5x3 creates too many anchors. anchor_encoder slows down the training significantly)

souryuu committed:
changed detail in config_v1.py
added recall and precision calculation
set test dataset to val2014
remove random picking data during test

souryuu committed:
fixed number of test data
fixed some evaluation problems

souryuu committed:
fix wrong sorting in sample_rpn_outputs

souryuu committed:
simplified config_v1 (no need to set _is_training)
revert partial sort (too buggy)

souryuu committed:
Change only_positive to False for training from scratch

souryuu committed:
excluded clowd instances from dataset
changed the layers for pyramid features C2-4 to match the original paper (endpoints of each resnet block)
test.py generates results.json file
pycocoEval.py evaluates AP and AR from results.json

***my max current AP 0.5 is 0.262

souryuu commented:
The one in fix_testing branch is included with evaluation API from coco.
However, I have change many things on the branch. I think you can not directly use it to evaluate your trained network. You may need to retrain the network.

souryuu committed:
failed version dont use this one

souryuu committed:
commit before rollback

souryuu committed:
roll back some part to v1

souryuu committed:
test.py should now able to test bounding box AP (mask is not included yet)
current mAP@50 IOU by FPN+Resnet50 (training from scratch) is 43.7%
changed anchor scale to match with the original paper
changed head parts of RPN RCNN and Mask
rename and remove multiple variables

souryuu committed:
added segmentation evaluation

souryuu committed:
fixed shuffle and queue during training
changed some hyper params to reduce overfitting (mAP@1M train:0.47 test:0.36)

souryuu committed:
fixed some memory issues (use script/train.sh for training)
removed images with extreme aspect ratio which cause OOM from tfrecords (records must be re-created)

souryuu commented:
With current version, I added the evaluation part using API from MSCOCO for both bounding box and mask. You can evaluate the network by 
```
python train/test.py
python pycocoEval.py
```
 After training up to 1M+ iteration, I got mAP(mask IoU=0.5) on training data = 0.47 and on testing data = 0.36 (comparing to original paper Resnet 50 mAP of 0.512). Please note that your previously trained networks and created tfrecords may not work with this version because I changed a lot of network settings.

I found 2 problems but I will not have time to fix it.
1). I tried to set is_training in Resnet and header to False during training but the performance dropped to ~0.20. If is_training is set to true, I got the above performance . I think this has something to do with batch normalization layers. Probably batch_size =1 is the problem. However, it seems that tensorflow does not support for batching images with dynamic shape.
2). When training with too high/low aspect ratio images, the convolutional features in pyramid become too large and GPU goes OOM. In order to train the current version, I removed too high/low aspect ratio images from the tfrecords.
